### PR TITLE
Add unit tests for tty module

### PR DESCRIPTION
**Pull request by constreference**
This PR addresses part of https://github.com/containers/youki/issues/8, by adding unit tests for the tty module.
It adds two dev-dependencies, tempfile for easing the creation of temporary directories and serial_test to force the tests to execute in serial, since we require setting the current working directory to the testdir for each test, since console-socket is expected to be in this current working directory

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,3 +26,5 @@ oci_spec = { version = "0.1.0", path = "./oci_spec" }
 [dev-dependencies]
 oci_spec = { version = "0.1.0", path = "./oci_spec", features = ["proptests"] }
 quickcheck = "1"
+serial_test = "0.5.1"
+tempfile = "3.2.0"
```
Comment (utam0k) [inline Cargo.toml:30]: How about moving the one used in the cgroups test to a more generic location and using it?
https://github.com/containers/youki/blob/1afde706fbfb94715f554f7fe7a31bb0eb86206c/src/cgroups/test.rs#L13

File: src/tty.rs [hunk 2]
```diff
@@ -68,3 +68,92 @@ pub fn setup_console(console_fd: FileDescriptor) -> Result<()> {
...
+        let cleanup = teardown(paths.0);
+        assert!(cleanup.is_ok());
+    }
+
+    #[test]
+    #[serial]
+    fn test_setup_console_socket_empty() {
+        let init =  setup();
+        assert!(init.is_ok());
+        let paths = init.unwrap();
```
Comment (utam0k) [inline src/tty.rs:122]: Other codes, too, but how about something like the following?
```suggestion
        let (testdir_path, rundir_path, socket_path) = init.unwrap();
```

File: Cargo.toml (hunk 1 above)
Comment (constreference) [reply Cargo.toml]: Moved the TempDir implmentation onto utils

File: src/tty.rs (hunk 2 above)
Comment (constreference) [reply src/tty.rs]: Changed it to the following

Comment (constreference) [reply src/tty.rs]: Thank you for reviewing

**Review by utam0k — APPROVED** [review]
lgtm

**Pull request merged**